### PR TITLE
[continuable] add a new port

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?

Add a new port for https://github.com/naios/continuable. There was no port request for this.

### Which triplets are supported/not supported? Have you updated the CI baseline?

Since this is a **header-only** library. All kinds of triplets must be available.

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

I will check it again :)

* Patching 
   * [x] Minimize patches

### CODE REVIEW

File: ports/continuable/fix-cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+-else()
+-  string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR}
+-                      CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
+-endif()
++# if (CTI_CONTINUABLE_IS_FIND_INCLUDED)
++#   set(CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT OFF)
++# else()
++#   string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR}
++#                       CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
++# endif()
```
Comment (NancyLi1013) [inline ports/continuable/fix-cmakelists.patch:19]: Please delete these lines directly or use `if(0)` to minimize the patch.

File: ports/continuable/fix-cmakelists.patch [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+-else()
++# if (CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
++#   include(cmake/CMakeLists.txt)
++#   add_subdirectory(dep)
++# else()
+   if(NOT TARGET function2::function2)
+     find_package(function2 4 REQUIRED)
+   endif()
+-endif()
++# endif()
```
Comment (NancyLi1013) [inline ports/continuable/fix-cmakelists.patch:39]: Please add `AND FALSE` or` && 0` to minimize the patch. See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#minimize-patches

File: ports/continuable/fix-cmakelists.patch [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ # continuable-base
+-if (CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
++# if (CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
+   add_library(continuable-base INTERFACE)
+-else()
+-  add_library(continuable-base INTERFACE IMPORTED GLOBAL)
+-endif()
++# else()
++#   add_library(continuable-base INTERFACE IMPORTED GLOBAL)
++# endif()
```
Comment (NancyLi1013) [inline ports/continuable/fix-cmakelists.patch:50]: Please delate these lines directly to minimize the patch.

File: ports/continuable/fix-cmakelists.patch [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ 
+-if (CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
++# if (CTI_CONTINUABLE_IS_TOP_LEVEL_PROJECT)
+   add_library(continuable INTERFACE)
+-else()
+-  add_library(continuable INTERFACE IMPORTED GLOBAL)
+-endif()
++# else()
++#   add_library(continuable INTERFACE IMPORTED GLOBAL)
++# endif()
```
Comment (NancyLi1013) [inline ports/continuable/fix-cmakelists.patch:66]: Please delete these lines directly to minimize the patch.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/continuable/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        -DCTI_CONTINUABLE_WITH_NO_EXCEPTIONS=OFF
+        -DCTI_CONTINUABLE_WITH_UNHANDLED_EXCEPTIONS=ON
+        -DCTI_CONTINUABLE_WITH_EXPERIMENTAL_COROUTINE=ON
+        -DCTI_CONTINUABLE_WITH_CPP_LATEST=OFF
+)
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
+
+file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/continuable RENAME copyright)
```
Comment (NancyLi1013) [inline ports/continuable/portfile.cmake:28]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
I also noticed that LICENSE.txt has been installed in CMakeLists.
https://github.com/Naios/continuable/blob/master/CMakeLists.txt#L171

Maybe we just need to update the destination.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/continuable/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+file(REMOVE ${CURRENT_PACKAGES_DIR}/ReadMe.md
```
Comment (NancyLi1013) [inline ports/continuable/portfile.cmake:29]:
```suggestion
file(REMOVE ${CURRENT_PACKAGES_DIR}/Readme.md
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
The failures on x64-linux are like this:

```
The following files are placed in
/mnt/vcpkg-ci/packages/continuable_x64-linux:

    /mnt/vcpkg-ci/packages/continuable_x64-linux/Readme.md

Files cannot be present in those directories.

Found 1 error(s). Please correct the portfile:
    /agent/_work/1/s/ports/continuable/portfile.cmake
```

**Comment by luncliff** [comment]
Thanks for the feedback @NancyLi1013. I applied your suggestions in 91ae74b

**Comment by ras0219-msft** [comment]
Thanks for the new port!

**Pull request merged**